test/commit-validator.spec.js

```diff
@@ -1,93 +1,259 @@
-      false,
+  it('follows valid revert format', async () => {
+    const { valid, errors, warnings } = await validator.validateCommit(
+      'Revert "refactor(foo): JIRA-1234 Extract method"',
     );
+
+    expect(valid).toBe(true);
+    expect(errors).toEqual([]);
+    expect(warnings).toEqual([]);
   });
 
-  it('returns false if message does not contain ticket number', async () => {
-    const validator = new CommitValidator(options);
+  describe('release', () => {
+    it('contains a prefix', async () => {
+      const { valid, errors, warnings } = await validator.validateCommit('Releasing v1.0.0');
+
+      expect(valid).toBe(true);
+      expect(errors).toEqual([]);
+      expect(warnings).toEqual([]);
+    });
+
+    it('does not include a prefix', async () => {
+      const { valid, errors, warnings } = await validator.validateCommit('Releasing 1.0.0');
+
+      expect(valid).toBe(true);
+      expect(errors).toEqual([]);
+      expect(warnings).toEqual([]);
+    });
+
+    it('is a release candidate', async () => {
+      const { valid, errors, warnings } = await validator.validateCommit('Releasing 1.0.0.rc-1');
 
-    expect((await validator.validateCommit('refactor(hello): Extract method')).valid).toBe(false);
+      expect(valid).toBe(true);
+      expect(errors).toEqual([]);
+      expect(warnings).toEqual([]);
+    });
+
+    it('is "Release" instead of "Releasing"', async () => {
+      const { valid, errors, warnings } = await validator.validateCommit('Release 1.0.0.rc-1');
+
+      expect(valid).toBe(true);
+      expect(errors).toEqual([]);
+      expect(warnings).toEqual([]);
+    });
   });
 
-  it('returns true if message follows valid format', async () => {
-    const validator = new CommitValidator(options);
+  describe('breaking change', () => {
+    it('follows valid format', async () => {
+      const { valid, errors, warnings } = await validator.validateCommit(
+        `
+        feat(bar): JIRA-1234 use new method
 
-    expect((await validator.validateCommit('refactor(foo): JIRA-1234 Extract method')).valid).toBe(
-      true,
-    );
-    expect((await validator.validateCommit('feat(bar): JIRA-1234 Add new feature')).valid).toBe(
-      true,
+        BREAKING CHANGE:
+        Use new method instead of old one.
+      `.trim(),
+      );
+
+      expect(valid).toBe(true);
+      expect(errors).toEqual([]);
+      expect(warnings).toEqual([]);
+    });
+
+    it('contains warning if no leading blank line', async () => {
+      const { valid, errors, warnings } = await validator.validateCommit(
+        `
+        feat(bar): JIRA-1234 use new method
+        BREAKING CHANGE:
+        Use new method instead of old one.
+      `.trim(),
+      );
+
+      expect(valid).toBe(true);
+      expect(errors).toEqual([]);
+      expect(warnings).toHaveLength(1);
+      expect(warnings).toStrictEqual([expect.objectContaining({ name: 'footer-leading-blank' })]);
```

Comment:
I used `toStrictEqual` in the tests to enforce order and to ensure no new rules added mess up the reported errors.